src/main/scala/com/cognite/sdk/scala/v1/sequenceRows.scala

```diff
@@ -12,45 +12,110 @@ final case class SequenceColumnSignature(
     valueType: String = "DOUBLE"
 )
 final case class SequenceRow(rowNumber: Long, values: Seq[Json])
-final case class SequenceRowsInsertById(id: Long, columns: Seq[String], rows: Seq[SequenceRow])
-final case class SequenceRowsInsertByExternalId(
-    externalId: String,
+final case class SequenceRowsInsert(
+    id: Option[Long],
+    externalId: Option[String],
     columns: Seq[String],
     rows: Seq[SequenceRow]
-)
-final case class SequenceRowsDeleteById(id: Long, rows: Seq[Long])
-final case class SequenceRowsDeleteByExternalId(externalId: String, rows: Seq[Long])
-sealed trait SequenceRowsQuery {
-  val start: Option[Long]
-  val end: Option[Long]
-  val limit: Option[Int]
-  val cursor: Option[String]
-  val columns: Option[Seq[String]]
-
-  def withCursorAndLimit(newCursor: Option[String], limit: Option[Int]): SequenceRowsQuery
+) {
+  require(id.isDefined || externalId.isDefined)
 }
-final case class SequenceRowsQueryById(
-    id: Long,
-    start: Option[Long],
-    end: Option[Long],
-    limit: Option[Int],
-    cursor: Option[String],
-    columns: Option[Seq[String]]
-) extends SequenceRowsQuery {
-  def withCursorAndLimit(newCursor: Option[String], limit: Option[Int]): SequenceRowsQuery =
-    this.copy(cursor = newCursor, limit = limit)
+object SequenceRowsInsert {
+  def apply(id: Long, columns: Seq[String], rows: Seq[SequenceRow]): SequenceRowsInsert =
+    SequenceRowsInsert(Some(id), None, columns, rows)
+
+  def apply(externalId: String, columns: Seq[String], rows: Seq[SequenceRow]): SequenceRowsInsert =
+    SequenceRowsInsert(None, Some(externalId), columns, rows)
+
+  def apply(
+      cogniteId: CogniteId,
+      columns: Seq[String],
+      rows: Seq[SequenceRow]
+  ): SequenceRowsInsert =
+    cogniteId match {
+      case id: CogniteInternalId =>
+        SequenceRowsInsert(id.id, columns, rows)
+      case id: CogniteExternalId =>
+        SequenceRowsInsert(id.externalId, columns, rows)
+    }
 }
-final case class SequenceRowsQueryByExternalId(
-    externalId: String,
+
+final case class SequenceRowsDelete(id: Option[Long], externalId: Option[String], rows: Seq[Long]) {
+  require(id.isDefined || externalId.isDefined)
+}
+object SequenceRowsDelete {
+  def apply(id: Long, rows: Seq[Long]): SequenceRowsDelete =
+    SequenceRowsDelete(Some(id), None, rows)
+
+  def apply(externalId: String, rows: Seq[Long]): SequenceRowsDelete =
+    SequenceRowsDelete(None, Some(externalId), rows)
+
+  def apply(
+      cogniteId: CogniteId,
+      rows: Seq[Long]
+  ): SequenceRowsDelete =
+    cogniteId match {
+      case id: CogniteInternalId =>
```

Comment:
Can probably do `case CogniteInternalId(id)` and `case CogniteExternalId(externalId)`?

Reply:
yes this look nicer, updated ;)